### PR TITLE
Fix not all lxd profiles applied on local charm upgrade

### PR DESCRIPTION
**Pull request by hmlanigan**
## Description of change

Upgrade of a local charm and a charm store charm follow different code paths in the uniter.
If upgrade via charmModified is due to local charm upgrade, check status of lxd profile 
upgrade before continuing.  Add corresponding unit tests.

## QA steps

Run assess_resources.py test.

1. bootstrap lxd
2. deploy bundle-no-sub.yaml from bug 1813044.
2. upgrade lxd-profile charm, watch all lxd machines start using the new profile version.

### CODE REVIEW

**Comment by hmlanigan** [comment]
[1813044 follow on: not all local charm profiles applied during upgrade](https://trello.com/c/hl5TdAP9/1039-1813044-follow-on-not-all-local-charm-profiles-applied-during-upgrade)

File: worker/uniter/resolver.go [hunk 1]
```diff
@@ -288,6 +288,18 @@ func (s *uniterResolver) nextOp(
 	// inform the uniter workers to run the upgrade hook.
 	if charmModified(localState, remoteState) {
 		if s.config.ModelType == model.IAAS {
+			if *localState.CharmURL != *remoteState.CharmURL {
```
Comment (SimonRichardson) [inline worker/uniter/resolver.go]: Interestingly I had this originally, but removed it because I couldn't get it to trigger when testing. How annoying!

**Comment by hmlanigan** [comment]
make-check-juju hit lp: 1793276

$$merge$$

**Comment by hmlanigan** [comment]
$$merge$$

**Pull request merged**